In [1]:
%load_ext autoreload
%autoreload 2

In [9]:
import pandas as pd
import geopandas as gpd 
import numpy as np

from sklearn import preprocessing

import matplotlib.pyplot as plt
import plotly.express as px

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
import warnings
warnings.filterwarnings("ignore")

In [3]:
import sys
sys.path.append("../") 

from utils.paths import make_dir_line

modality = 'u'
project = 'nlp_course_w'
data = make_dir_line(modality, project)

raw = data('raw')

# AIMS

In [6]:
sms = pd.read_table(raw / 'sms+spam+collection/SMSSpamCollection', header=None)
sms.head()

0                                                  1
0   ham  Go until jurong point, crazy.. Available only ...
1   ham                      Ok lar... Joking wif u oni...
2  spam  Free entry in 2 a wkly comp to win FA Cup fina...
3   ham  U dun say so early hor... U c already then say...
4   ham  Nah I don't think he goes to usf, he lives aro...

In [7]:
sms.describe()

0                       1
count   5572                    5572
unique     2                    5169
top      ham  Sorry, I'll call later
freq    4825                      30

## Data Info 

Here, we have a collection of text data known as a corpus. Specifically, there are 5,572 SMS messages written in English, serving as training examples. The first column is the target variable containing the class labels, which tells us if the message is spam or ham (aka not spam). The second column is the SMS message itself, stored as a string.

In [8]:
y = sms[0]
y.value_counts()

0
ham     4825
spam     747
Name: count, dtype: int64

It looks like there are far fewer training examples for spam than ham—we'll take this imbalance into account in the analysis.

In addition, we need to encode the class labels in the target variable as numbers to ensure compatibility with some models in Scikit-learn. Because we have binary classes, let's use LabelEncoder and set 'spam' = 1 and 'ham' = 0.

In [11]:
le = preprocessing.LabelEncoder()
y_enc = le.fit_transform(y)
y_enc

array([0, 0, 1, ..., 0, 0, 0])

Next, we place the SMS message data into its own table. We must convert this corpus into useful numerical features so we can train this classifier and this is where NLP works its magic!

In [12]:
raw_text = sms[1]
raw_text

0       Go until jurong point, crazy.. Available only ...
1                           Ok lar... Joking wif u oni...
2       Free entry in 2 a wkly comp to win FA Cup fina...
3       U dun say so early hor... U c already then say...
4       Nah I don't think he goes to usf, he lives aro...
5       FreeMsg Hey there darling it's been 3 week's n...
6       Even my brother is not like to speak with me. ...
7       As per your request 'Melle Melle (Oru Minnamin...
8       WINNER!! As a valued network customer you have...
9       Had your mobile 11 months or more? U R entitle...
10      I'm gonna be home soon and i don't want to tal...
11      SIX chances to win CASH! From 100 to 20,000 po...
12      URGENT! You have won a 1 week FREE membership ...
13      I've been searching for the right words to tha...
14                    I HAVE A DATE ON SUNDAY WITH WILL!!
15      XXXMobileMovieClub: To use your credit, click ...
16                             Oh k...i'm watching here:)
17      Eh u r

In [14]:
pd.isnull(sms).sum()

0    0
1    0
dtype: int64

In [5]:
print('Ok_')

Ok_
